build.gradle

```diff
@@ -699,38 +691,40 @@ subprojects { Project subproject ->
         relocate "com.github.benmanes.caffeine", "io.micronaut.caffeine"
         relocate "org.objectweb.asm", "io.micronaut.asm"
     }
-    
-    shadowJar.enabled = false
 
     def isDocumented = subproject.name != 'inject' && subproject.name != 'core' && subproject.name != 'aop' && !subproject.name.contains('asciidoc-config-props')
 
     dependencies {
         documentation "org.codehaus.groovy:groovy-templates:$groovyVersion"
         documentation "org.codehaus.groovy:groovy-dateutil:$groovyVersion"
         if (subproject.name != "bom") {
-            compile dependencyVersion("slf4j")
+            api dependencyVersion("slf4j")
         }
 
 
         if (isDocumented) {
             compileOnly "io.micronaut.docs:micronaut-docs-asciidoc-config-props:$micronautDocsVersion"
         }
         compileOnly "com.github.ben-manes.caffeine:caffeine:$caffeineVersion"
-        testCompile "com.github.ben-manes.caffeine:caffeine:$caffeineVersion"
-        testCompile dependencyVersion("groovy")
-        testCompile(dependencyVersion("spock")) {
+        testImplementation "com.github.ben-manes.caffeine:caffeine:$caffeineVersion"
+        testImplementation dependencyVersion("groovy")
+        testImplementation(dependencyVersion("spock")) {
             exclude module: 'groovy-all'
         }
-        testCompile "cglib:cglib-nodep:3.3.0"
-        testCompile "org.objenesis:objenesis:1.4"
+        testImplementation "cglib:cglib-nodep:3.3.0"
+        testImplementation "org.objenesis:objenesis:1.4"
 
-        testRuntime "ch.qos.logback:logback-classic:1.2.3"
-        testCompile "org.codehaus.groovy:groovy-test:$groovyVersion"
+        testRuntimeOnly "ch.qos.logback:logback-classic:1.2.3"
+        testImplementation "org.codehaus.groovy:groovy-test:$groovyVersion"
         compileOnly "org.ow2.asm:asm:$asmVersion"
         compileOnly "org.ow2.asm:asm-commons:$asmVersion"
+        annotationProcessor "org.ow2.asm:asm:$asmVersion"
+        annotationProcessor "org.ow2.asm:asm-commons:$asmVersion"
```

Comment:
Do these need to be both annotationProcessor and compileOnly? Is it compileOnly for shadowjar?

Reply:
I believe `comileOnly` is needed for groovy